cmd/argocd/commands/app.go

```diff
@@ -1305,6 +1312,17 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 				diffOption.res = res
 				diffOption.revisions = revisions
 				diffOption.sourcePositions = sourcePositions
+			} else if compareDesire {
+				q := application.ApplicationManifestQuery{
+					Name:         &appName,
+					Revision:     &revision,
+					AppNamespace: &appNs,
+				}
+				res, err := appIf.GetManifests(ctx, &q)
+				errors.CheckError(err)
+				diffOption.res = res
+				diffOption.revision = revision
+				diffOption.compareDesire = true
```

Comment:
Why not just setting `diffOption.compareDesired = compareDesired` on all existing `ApplicationManifestQuery`?

Reply:
Yes, resolved this one.